src/providers/ec2/ec2.go

```diff
@@ -101,3 +145,45 @@ func (p provider) ShouldRetry() bool {
 func (p *provider) BackoffDuration() time.Duration {
 	return util.ExpBackoff(&p.backoff, maxBackoff)
 }
+
+func fetchBody(client *http.Client, url string, acceptedStatuses ...int) ([]byte, int, error) {
```

Comment:
capturing OOB discussion:
just return the status, lose the acceptedStatuses stuff.